src/Components/WebAssembly/WebAssembly.Authentication/src/Services/AuthorizationMessageHandler.cs

```diff
@@ -36,6 +37,12 @@ public class AuthorizationMessageHandler : DelegatingHandler
         {
             _provider = provider;
             _navigation = navigation;
+
+            // Invalidate the cached _lastToken when the authentication state changes
+            if (_provider is AuthenticationStateProvider authStateProvider)
+            {
+                authStateProvider.AuthenticationStateChanged += _ => { _lastToken = null; };
+            }
```

Comment:
Looks great, however you need to implement IDisposable to unsubscribe from the event, otherwise this results in a memory leak over time.

Reply:
https://github.com/dotnet/aspnetcore/pull/36358/commits/8d9a5cc61834c658e3b4046fe14c4ac328c62ba3

Will require an API change though. If I can get an approval for the PR I'll look into getting API approval. Given the minor API change, I believe it shouldn't be an issue.

Reply:
Actually, we may be able to get around the public API by doing https://github.com/dotnet/aspnetcore/pull/36358/commits/e2351d3df4c1fb12eeeeaa1d3559f40392a1fc2d. Thoughts?

Reply:
(Discussed offline, good to merge now)